app/v1/controllers/requestParsers/validators/AmendSelfEmploymentAnnualSummaryValidator.scala

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright 2020 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1.controllers.requestParsers.validators
+
+import v1.controllers.requestParsers.validators.validations._
+import v1.models.errors._
+import v1.models.request.amendAnnualSummary._
+
+class AmendSelfEmploymentAnnualSummaryValidator extends Validator[AmendAnnualSummaryRawData] {
+
+  private val validationSet = List(parameterFormatValidation, parameterRuleValidation, bodyFormatValidation, bodyFieldValidation)
+
+  private def parameterFormatValidation: AmendAnnualSummaryRawData => List[List[MtdError]] = (data: AmendAnnualSummaryRawData) => {
+    List(
+      NinoValidation.validate(data.nino),
+      BusinessIdValidation.validate(data.businessId),
+      TaxYearValidation.validate(data.taxYear)
+    )
+  }
+
+  private def parameterRuleValidation: AmendAnnualSummaryRawData => List[List[MtdError]] = { data =>
+    List(
+      TaxYearNotSupportedValidation.validate(data.taxYear)
+    )
+  }
+
+  private def bodyFormatValidation: AmendAnnualSummaryRawData => List[List[MtdError]] = { data =>
+    List(
+      JsonFormatValidation.validate[AmendAnnualSummaryBody](data.body)
```

Comment:
Validator doesn't seem to cover for when the adjustments, nonFinancials and allowances objects are included but empty.